### DISCUSSION TITLE
Linkerd Helm chart installation certificate issue

### DISCUSSION THREAD

**Discussion by shahriak** [linkerd/linkerd2#5094] [Q&A]
Greetings Linkerd Community,

We are trying to install Linkerd using Helm. We have an automated process which simply pulls the Helm chart and deploys it onto an AKS cluster. The only hurdle is that the Helm command requires root and issuer certs to be passed. I would like to reduce the command from 1 to 2

I have already looked into cert-manager but it seems rather complicated. Could you guys please provide some input?

**Is it possible to update the certs after linkerd installation using Helm and restart the pods to pick up the certs?**

1)
```
	helm install linkerd2 \
	  --set-file global.identityTrustAnchorsPEM=ca.crt \
	  --set-file identity.issuer.tls.crtPEM=issuer.crt \
	  --set-file identity.issuer.tls.keyPEM=issuer.key \
	  --set identity.issuer.crtExpiry=$exp \
	  linkerd/linkerd2

```
2)
`	helm install linkerd2 linkerd/linkerd2
`

**Comment by grampelberg** [comment]
This is an unfortunate limitation of helm. We can generate certificates with helm, but not persist them between upgrades. If you want a single command I would recommend using `linkerd install`.

**Comment by shahriak** [comment]
There is a way to bypass this limitation. I was able to include some dummy values for the 4 parameters and it was able to install all the components but the pods are in "CrashLoopBackOff" state which makes sense since the certs are invalid. How do we go about updating the certs once Linkerd has been installed?

**Reply by grampelberg** [reply]
`helm upgrade` would be the correct command.

**Reply by shahriak** [reply]
Ok, the problem we have is, we have an e2e automation which basically pulls the charts from an ACR and runs Helm command to install charts. So the charts are packaged up and intact and we cannot make any modifications to the charts after deployment.
We can update the cluster components to have some sort of logic to update the identity certs. So, does it make sense to modify the helm charts to have an Init container running inside each pods that basically pulls secret certs from a central keyvault and puts them in the configs?

**Reply by grampelberg** [reply]
What problem are you trying to solve to begin with? I'm super confused.

**Reply by shahriak** [reply]
Apologies for not being too clear. What I am trying to do is basically update the root and issuer cert once linkerd has been deployed in the cluster and restart all pods to use the latest certs and bring the pods to a "Running" state. Does that make sense?

**Reply by grampelberg** [reply]
I guess I don't understand why you want to update them after the install?

**Reply by shahriak** [reply]
The problem is during helm installation, I cannot pass the arguments "--set file" to pass the certificates (it is owned by another team and is not our control). The only way I am able to bypass this is update the chart with some dummy certs and then after installation is complete, update the root, issuer and key with proper certs.

**Reply by grampelberg** [reply]
Why can't you pass `--set`?

**Comment by feiyushi** [comment]
@shahriak Do you have a solution now? I'm interested to know how to helm upgrade with new certs with no down time. thx
